website/pages/docs/nia/network-drivers.mdx

```diff
@@ -6,10 +6,49 @@ description: >-
+
+Autogenerated root modules for tasks are stored in local subdirectories of the Terraform working directory. By default, the working directory `sync-tasks` is created in the current directory. To configure where Terraform configuration files are stored, set [`working_dir`](/docs/nia/installation/configuration#working_dir) for the Terraform driver to the desired path.
+
+```
+sync-tasks/
+  my-task/
+    main.tf
+    variables.tf
+    terraform.tfvars
+    terraform.tfvars.tmpl
+```
+
+The following files of the root module are generated for each task. An [example of a root module created by Consul-Terraform-Sync](https://github.com/hashicorp/consul-terraform-sync/tree/master/examples) can be found in the project repository.
+* `main.tf` - The main file contains the terraform block, provider blocks, and a module block calling the module configured for the task.
+    * `terraform` block - The corresponding provider source and versions for the task from the configuration files are placed into this block for the root module. The Terraform backend from the configuration is also templated here.
+    * `provider` blocks - The provider blocks generated in the root module resemble the provider blocks in the configuration. They have identical arguments present nad are set from the intermediate variable created per provider.
+    * `module` block - The module block is where the task's module is called as a [child module](https://www.terraform.io/docs/configuration/modules.html#calling-a-child-module). Required and optional input variables are passed as arguments to the module.
+* `variables.tf` - This file contains 2 types of variable defintions. The required `services` input variable which determines module compatibility with Consul-Terraform Sync (read more on [compatible Terraform modules](#compatible-terraform-modules) for more details) and various intermediate variables used to dynamically configure providers. These intermediate provider variables are interpolated from the provider blocks and arguments configured in the Consul-Terraform-Sync configuration.
+* `variables.module.tf` - This file is conditionally created if there are [variables configured for the task](/docs/nia/installation/configuration#variable_files) and contains the interpolated variable declarations that match the variables from configuration. These are then used to proxy the configured variables to the module through explicit assignment in the module block.
+* `terraform.tfvars` - The variable definitions file is where the services input variable is assigned values from the Consul Catalog. It is periodically updated to reflect the current state of the configured set of services for the task.
```

Comment:
Just want to make sure I'm on the right page, it seems like both `.tf` and `.tfvars` files are called variable definitions files. is that right?

Reply:
To be honest I'm not confident and I've been heavily relying on the Terraform docs. `.tfvars` is variable definitions files ([source](https://www.terraform.io/docs/configuration/variables.html#variable-definitions-tfvars-files)), and from my understanding the `variables.tf` is just convention to centrally locate variable declarations.

It looks like I misused "definitions" in the `variables.tf` section, updated to "This file contains 2 types of variable declarations".